ext/date/php_date.c

```diff
@@ -172,19 +172,22 @@ static HashTable *date_object_get_properties_for(zend_object *object, zend_prop_
 static HashTable *date_object_get_gc_interval(zend_object *object, zval **table, int *n);
 static HashTable *date_object_get_properties_interval(zend_object *object);
 static HashTable *date_object_get_gc_period(zend_object *object, zval **table, int *n);
-static HashTable *date_object_get_properties_period(zend_object *object);
 static HashTable *date_object_get_properties_for_timezone(zend_object *object, zend_prop_purpose purpose);
 static HashTable *date_object_get_gc_timezone(zend_object *object, zval **table, int *n);
 static HashTable *date_object_get_debug_info_timezone(zend_object *object, int *is_temp);
 static void php_timezone_to_string(php_timezone_obj *tzobj, zval *zv);
 
+static void create_date_period_datetime(timelib_time *datetime, zend_class_entry *ce, zval *zv);
```

Comment:
Please move these helper functions to before the location where they are used, and then hence you don't have to forward declare them any more.